src/cdk-experimental/menu/menu-item.ts

```diff
@@ -35,21 +50,112 @@ export class CdkMenuItem {
   }
   private _disabled = false;
 
+  /**
+   * If this MenuItem is a regular MenuItem, outputs when it is triggered by a keyboard or mouse
+   * event.
+   */
+  @Output('cdkMenuItemTriggered') triggered: EventEmitter<void> = new EventEmitter();
+
   constructor(
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    @Inject(CDK_MENU) private readonly _parentMenu: Menu,
+    private readonly _dir: Directionality,
     /** Reference to the CdkMenuItemTrigger directive if one is added to the same element */
     @Self() @Optional() private readonly _menuTrigger?: CdkMenuItemTrigger
   ) {}
 
-  /** Open the menu if one is attached */
+  /** Place focus on the element. */
+  focus() {
+    this._elementRef.nativeElement.focus();
+  }
+
+  /**
+   * If the menu item is not disabled and the element does not have a menu trigger attached, emit
+   * on the cdkMenuItemTriggered emitter and close all open menus.
+   */
   trigger() {
-    if (!this.disabled && this.hasMenu()) {
-      this._menuTrigger!.toggle();
+    if (!this.disabled && !this.hasMenu()) {
+      this.triggered.next();
+      this._getMenuStack().closeAll();
     }
   }
 
   /** Whether the menu item opens a menu. */
   hasMenu() {
-    return !!this._menuTrigger && this._menuTrigger.hasMenu();
+    return !!this._menuTrigger?.hasMenu();
+  }
+
+  /** Return true if this MenuItem has an attached menu and it is open. */
+  isMenuOpen() {
+    return !!this._menuTrigger?.isMenuOpen();
+  }
+
+  /**
+   * Get a reference to the rendered Menu if the Menu is open and it is visible in the DOM.
+   * @return the menu if it is open, otherwise undefined.
+   */
+  getMenu(): Menu | undefined {
+    return this._menuTrigger?.getMenu();
+  }
+
+  /** Get the MenuItemTrigger associated with this element. */
+  getMenuTrigger(): CdkMenuItemTrigger | undefined {
+    return this._menuTrigger;
+  }
+
+  /** Get the label for this element which is required by the FocusableOption interface. */
+  getLabel(): string {
+    // TODO(andy): implement a more robust algorithm for determining nested text
+    return this._elementRef.nativeElement.innerText;
```

Comment:
`innerText` will trigger a reflow, because it has some logic where it excludes hidden text. We should use `textContent` here.